wifiphisher/common/attacks.py

```diff
@@ -0,0 +1,322 @@
+from wifiphisher.common.constants import *
+import
+import time, threading, socket, fcntl, struct, subprocess
+
+
+class Manage(object):
+
+    """
+    This class is used to determine which attack has been chosen.
+    If the attack isn't valid it will throw an exception :
+    NotValidAttack
+    """
+
+    def __init__(self, attack):
+
+        """
+        Setup the class with all the given arguments.
+
+        :param self: A Manage object.
+        :param attack: Which attack has been chosen.
+        :type self: Manage
+        :type attack: string
+        :return: None
+        """
+
+        self.attack = attack
+        self.attacks = ["wps_pbc"]
+        self.log_file = "/tmp/wifiphisher-attacks.tmp"
+        self.check()
+
+    def check(self):
+
+        """
+        Check which attack has been chosen, if the attack
+        is not valid raise NotValidAttack()
+        """
+
+        if not self.attack in self.attacks :
+            raise NotValidAttack(self.attack, self.attacks)
+
+    def provide(self):
+
+        """
+        Returns what is needed for the attack to run
+
+        :param self: A Manage object
+        :type self: Manage
+        :return: what is needed for the attack to run
+        :rtype: list
+        """
+
+        self.needs = "all"
+        if self.attack == "wps_pbc":
+            #pay attention to this order when you create an attack because this is the
+            #the order that you're gonna recieve as parameter in your attack.
+            #see WPS_PBC as example
+            self.needs = ["deauth-object", "deauth_iface", "target_bssid", "network_manager"]
```

Comment:
@sophron These are all the objects, variables that are needed for the attack so function